core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/returnfirstnonnull/ReturnFirstNonNullUnitTest.java

```diff
@@ -0,0 +1,108 @@
+package returnfirstnonull;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.lang3.ObjectUtils;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Predicates;
+import com.google.common.collect.Iterables;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ReturnFirstNonNullUnitTest {
+
+    private List<Object> objects;
+
+    @Before
+    public void init() {
+        objects = Arrays.asList(
+            null,
+            new Object(),
+            new Object()
+        );
+    }
+
+    @Test
+    public void givenListOfObjects_whenIteratingWithForLoop_thenReturnFirstNonNull() {
+        Object object = null; 
+        for(int i = 0; i < objects.size(); i++) {
+            if(objects.get(i) != null) {
+                object = objects.get(i);
+                break;
+            }
+        }
+
+        assertTrue(object != null);
+        assertEquals(objects.get(1), object);
```

Comment:
As mentioned elsewhere, I think the tests might be clearer if we aimed for something like this:
```suggestion
        assertEquals("first non null", object);
```

Rather than the `objects.get(1)` reference. I can understand once at the beginning that we are working with an example where the first non null element is a string "first non null" and then all the assertions can be about that.


Happy to be convinced otherwise if you prefer your original approach.

Reply:
Agree with all the suggestions and approach. Changed